pkg/controller/mpi_job_controller_test.go

```diff
@@ -751,6 +765,209 @@ func TestShutdownWorker(t *testing.T) {
 	f.run(getKey(mpiJob, t))
 }
 
+func TestCreateSuspendedMPIJob(t *testing.T) {
+	impls := []kubeflow.MPIImplementation{kubeflow.MPIImplementationOpenMPI, kubeflow.MPIImplementationIntel}
+	for _, implementation := range impls {
+		t.Run(string(implementation), func(t *testing.T) {
+			f := newFixture(t)
+
+			// create a suspended job
+			var replicas int32 = 8
+			mpiJob := newMPIJob("test", &replicas, nil, nil)
+			mpiJob.Spec.RunPolicy.Suspend = pointer.Bool(true)
+			mpiJob.Spec.MPIImplementation = implementation
+			f.setUpMPIJob(mpiJob)
+
+			// expect creation of objects
+			scheme.Scheme.Default(mpiJob)
+			f.expectCreateServiceAction(newWorkersService(mpiJob))
+			cfgMap := newConfigMap(mpiJob, replicas)
+			updateDiscoverHostsInConfigMap(cfgMap, mpiJob, nil)
+			f.expectCreateConfigMapAction(cfgMap)
+			secret, err := newSSHAuthSecret(mpiJob)
+			if err != nil {
+				t.Fatalf("Failed creating secret")
+			}
+			f.expectCreateSecretAction(secret)
+			if implementation == kubeflow.MPIImplementationIntel {
+				f.expectCreateServiceAction(newLauncherService(mpiJob))
+			}
+
+			// expect creating of the launcher
+			fmjc := f.newFakeMPIJobController()
+			launcher := fmjc.newLauncherJob(mpiJob)
+			launcher.Spec.Suspend = pointer.Bool(true)
+			f.expectCreateJobAction(launcher)
+
+			// expect an update to add the conditions
+			mpiJobCopy := mpiJob.DeepCopy()
+			mpiJobCopy.Status.ReplicaStatuses = map[common.ReplicaType]*common.ReplicaStatus{
+				common.ReplicaType(kubeflow.MPIReplicaTypeLauncher): {},
+				common.ReplicaType(kubeflow.MPIReplicaTypeWorker):   {},
+			}
+			msg := fmt.Sprintf("MPIJob %s/%s is created.", mpiJob.Namespace, mpiJob.Name)
+			updateMPIJobConditions(mpiJobCopy, common.JobCreated, v1.ConditionTrue, mpiJobCreatedReason, msg)
+			updateMPIJobConditions(mpiJobCopy, kubeflow.JobSuspended, v1.ConditionTrue, mpiJobSuspendedReason, "MPIJob suspended")
+			msg = fmt.Sprintf("MPIJob %s/%s is suspended.", mpiJob.Namespace, mpiJob.Name)
+			updateMPIJobConditions(mpiJobCopy, common.JobRunning, v1.ConditionFalse, mpiJobSuspendedReason, msg)
+			f.expectUpdateMPIJobStatusAction(mpiJobCopy)
```

Comment:
expect zero workers

Reply:
The code above checks that already:
```golang
			mpiJobCopy.Status.ReplicaStatuses = map[common.ReplicaType]*common.ReplicaStatus{
				common.ReplicaType(kubeflow.MPIReplicaTypeLauncher): {},
				common.ReplicaType(kubeflow.MPIReplicaTypeWorker):   {},
			}
```
It does not specify `Active`, meaning it checks the value is 0.

The workers were never created in this scenario, so I cannot assert on delete actions.

Reply:
but you can assert that there were no pods were created

the status is not the same as pods being created, necessarily

Reply:
oh I see, so it's implicitly checked